demo/kotlin/src/main/kotlin/JupiterMoons.kt

```diff
@@ -49,10 +49,10 @@ internal fun `Jupiter moons demo`(time: Time): Int {
     // a different time. This is a manual override to work around a safety check.
 
     printBody("Jupiter",  jv)
-    printBody("Io",       jv + jm.moon[0].position().withTime(jv.t))
-    printBody("Europa",   jv + jm.moon[1].position().withTime(jv.t))
-    printBody("Ganymede", jv + jm.moon[2].position().withTime(jv.t))
-    printBody("Callisto", jv + jm.moon[3].position().withTime(jv.t))
```

Comment:
maybe better to hide moon anyway? Maybe it should've be named `moons` instead `moon`

Reply:
I like this change, and it matches what I did in the Java code. About hiding `moon`, I'm not sure, because there are cases where it could be nicer for the caller to iterate through them with a loop. And there is always the debate about whether a collection name should be singular or plural.

Reply:
I thought about it some more, and now I'm thinking you are right: it would be better to expose only the named fields for the moons, not the array. In practice, developers need to know which moon is which, not iterate through them generically. I will change this across all languages, update the documentation, and fix resulting breakage in the demos/tests. This is of course a breaking change for released code, but I'm willing to make an exception because I doubt many people are using the Jupiter moons functions yet.